java/spark/src/test/java/com/lancedb/lance/spark/write/InternalRowWriterArrowReaderTest.java

```
@@ -0,0 +1,99 @@
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.VectorUnloader;
+import org.apache.arrow.vector.ipc.message.ArrowRecordBatch;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.catalyst.expressions.GenericInternalRow;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class InternalRowWriterArrowReaderTest {
+
+  @Test
+  public void testInternalRowWriterArrowReader() throws InterruptedException, IOException {
+    try (BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE)) {
+      Field field = new Field("column1", FieldType.nullable(org.apache.arrow.vector.types.Types.MinorType.INT.getType()), null);
+      Schema schema = new Schema(Collections.singletonList(field));
+
+      final int totalRows = 125;
+      final int batchSize = 34;
+      final InternalRowWriterArrowReader arrowReader = new InternalRowWriterArrowReader(allocator, schema, batchSize);
+
+      AtomicInteger rowsWritten = new AtomicInteger(0);
+      AtomicInteger rowsRead = new AtomicInteger(0);
+      AtomicLong expectedBytesRead = new AtomicLong(0);
+
+      Thread writerThread = new Thread(() -> {
+        try {
+          for (int i = 0; i < totalRows; i++) {
+            InternalRow row = new GenericInternalRow(new Object[]{rowsWritten.incrementAndGet()});
+            arrowReader.write(row);
+            Thread.sleep(1);
+          }
+          arrowReader.setFinished();
+        } catch (Exception e) {
+          e.printStackTrace();
```

This one swallows the exception?